js/utils/helpers.js

```diff
@@ -2180,11 +2197,11 @@
                             }
 
                             if(captionHTML.trim().length && flag && posTop){
-                                html +=  "<figure><figcaption>"+captionHTML+ "</figcaption>" + videoHTML + videoClass +">"+ srcHTML +"</video></figure>" ;
+                                html +=  "<figure><figcaption>"+captionHTML+ "</figcaption>" + infoHTML + videoHTML + videoClass +">"+ srcHTML +"</video></figure>" ;
```

Comment:
In the iframe implementation, you're always attaching this at the beginning of the computed HTML (regardless of position attribute). Why are you not doing the same for video?

Reply:
That is because, `<iframe>` implementation has html written this way : 
// Encapsulate the iframe inside a figure tag
`html = '<figure class="embed-block ' + module._classNames.postLoad + (iframeClass.length ? (" "  + iframeClass): "") + '" style="' + (iframeStyle.length ? (" "  + iframeStyle ) : "") + '">' + html + "</figure>";`

So irrespective of the position, the `<span>` element is always inside the `<figure>`. However for the <video> this is not the case. Adding the `<span>` irrespective of position might cause the `<span>` to be defined outside the <figure>. That is why I had defined it that way.